### ISSUE TITLE
TDTM Filtering not working correctly when using Filter Field of Boolean type

### ISSUE THREAD

**Issue by jacebryan** [SalesforceFoundation/HEDAP#465]
**Issue identified by** [Brian Petersen (Queen's School of Business) ](https://powerofus.force.com/0058000000ABNmU) on July 13, 2017
**PUB Thread**: https://powerofus.force.com/_ui/core/chatter/groups/GroupProfilePage?g=0F980000000D6uj&fId=0D58000003VpuaU

**Issue**: TDTM Filtering not working correctly when using Filter Field of Boolean typeValue

**Steps to reproduce**:

1. Create a custom Apex Class that updates a Contact field value if some condition is met.
- Example: IF Military Service == True then add "Please Contact VA Services" in the Military Background field

2. Create a new Trigger Handler that references the following:
- Class = class created in Step 1
- Object = Contact
- Trigger Action = Before Insert; Before Update
- Filter Field = ex: hed__Deceased__c
- Filter Value = ex: 'True' OR 'true'

3. Test the Class and Trigger Handler by Creating a new Contact
- Create a new Contact and set the value of hed_Military_Service__c to TRUE
- Save the record
- The hed_Military_Background__c field now has the value of "Please contact VA Services"

_Note_: The Trigger Handler indicates that unless the Filter Field (i.e. hed_Deceased__C) and Filter Value (i.e True or true) are met then the Class referenced will not be called.

**The class should not be called unless the Trigger Handler condition is TRUE.**

**Comment by davidjray** [comment]
**lurch: add

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x80000007O6ofAAC" target="_blank">W-019578</a>

**Comment by ceiroa** [comment]
@weizsforg Would you mind adding some details about what you found to be the actual problem? Thank you!

**Comment by weizsforg** [comment]
I did a investigation in one of the HEDA trial org and I found the following: 
1. CheckBox (HEDA field): **doesn't work**
2. Picklist (HEDA field):**doesn't work** 
3. Text (HEDA field):**doesn't work** 
4. CheckBox (HEDA field without namespace: For example user put Military_Background__c but not hed__Military_Background__c): **works**
5. CheckBox (user's custom field):**doesn't work** 
6. Picklist (user's custom field):**doesn't work** 
7. Text (user's custom field):**doesn't work** 
8. Text (standard, like LastName): **works**
9. Checkbox (standard, like Email Opt Out): **works**

In summary:
1. Filter on standard field always works
2. Filter on HEDA field only works when you omit the namespace 
3. Filter on custom field outside HEDA will not work
4. The issue is not just for Boolean field type but all filed type

**Comment by mrbelvedere** [comment]
Included in production release rel/1.46
